### DISCUSSION TITLE
What do you use the OLED screen on your node for?

### DISCUSSION THREAD

**Discussion by thatSFguy** [meshtastic/Discussions#159] [Firmware]
Many nodes include an OLED screen, such as the Heltec, TLORA,. techo, tbeam, etc.  
**2 Questions:**
- What do you use the screen for?  
- What would you like to see on the screen,that it doesn't show today?

**Comment by kiam001** [comment]
i would like to be able to see my heltec stick v3 display which is not yet supported

**Reply by thatSFguy** [reply]
Hey @kiam001, if it did work, what would use the display for?  (besides getting the Bluetooth pairing code, to sync with your phone)

**Reply by kiam001** [reply]
Hi i use the https://heltec.org/project/wireless-stick-v3/ this have a Onboard 0.49-inch ```64*32``` dot matrix OLED display, if i use the WiFi LoRa 32(V3) Firmware display turn on but only with the  64*32 dot matrix so the most is missing, its recommend than to install the "wireless-stick-lite-v3" firmware so display dont turn on and the pin is default fixed on 123456 thats i edit, to own after setup.

if i use the WiFi LoRa 32(V3) Firmware need setup via web serial and change to fix pin and only whan i use the option Flip Screen on one site i see the battary state but this is usless for me.

**Reply by thatSFguy** [reply]
@kiam001 sounds like you need a separate thread, probably an "[issue](https://github.com/meshtastic/firmware/issues)", to troubleshoot this. In this tread I'm trying to discuss if anyone is using the screens, besides for getting the pin.  This discussion area is a replacement for discourse.  Good luck with getting your issue resolved!

**Comment by madeofstown** [comment]
At first I thought they would be useful for a quick glance at messages or battery status but I can just use the app for that. I have stopped considering nodes with screens as I feel they don't serve much purpose other than to drain battery.

**Reply by thatSFguy** [reply]
I agree.  The screen has lost value for the folks that understand how to ensure that their phone is connected via Bluetooth to their node.

**Reply by kalestew** [reply]
This is exactly why I think most users would be best served through a new "homepage" format for the display. By default, a single page that displays things like RF activity, last packet SNR/RSSI + time, battery status, bluetooth/wifi connection, GPS status, and small graphs/indicators for other useful statistics or metrics would make the display more useful for nodes that mostly just connect to apps. Button presses can page to more detailed statistics and/or the traditional node info displays.

**Reply by thatSFguy** [reply]
I started giving away window nodes that people can stick in their window and help with coverage in certsin areas while being unobtrusive.  Most will probably never connect to the node, but just a screen that shows the status would be helpful, since some nodes need an occasional reboot.

**Comment by easytarget** [comment]
For a remote and/or solar nodes, unplug the screen or buy a screenless device and enjoy the low power drain.

For permanently powered home nodes a screen showing status etc is useful. If you have a Wifi or BT connection being able to view messages is probably irrelevant, but seeing the network status, radio status, throughput &  connected nodes summary etc.. would be very nice. Especially the wifi status :-)

A lot of devices _do_ have screens, and the current attitude of 'provide basic info if screen present, but do not make screen mandatory' seems sensible. My office node has a E-Paper display, it's nice to just look at it to see if anything interesting is happening. My portable node is a T114, it has a hi-res hi-color TFT display. Not all screens are OLED these days.

AFIK there are GUI changes in the pipeline; so I'm in wait-and-see mode.

**Reply by thatSFguy** [reply]
For my solar nodes, I'm using the nRF52 RAK nodes without a screen.  The Android app has come such a long way with so many features, that it's hard to live without anywhere from node configuration to troubleshooting ;)

On the flip side, you need a basic understanding of how the node connects to the phone and making sure it is actually connected when you are sitting in your living room and the node is in the bedroom, etc.  Not a problem for me, but less techie people it is, so just being able to view messages on the node would be definite plus.  One guy in the local mesh named is node "Mark's Pager", which I thought was an interesting analogy :-)

I wasn't aware of UI changes coming,  I thought the changes were geared toward the T-Deck, which I'm running the experimental UI on.  Hopefully the updated screens, will have a "simple" version for less techie folks.

**Reply by kalestew** [reply]
For my eink nodes I have around my desk, a messages list would be a very welcome 'app page' to have for the standard display. In a large urban mesh, all the node info pages provide little value and displaying only the last message is just a waste of beautiful eink real-estate.

**Comment by kalestew** [comment]
I have always wanted a better realtime visualization of what the node is receiving or transmitting; making channel utilization more prominent on the display, across all views would be really cool. I was thinking that using the outermost pixels on some of the edges to reflect the 15s/1min/15min channel utilization and maybe an edge to reflect airtime with a simple animation occurring during tx. 

I just recently got started with Reticulum network Rnodes using some extra Heltec V3s I had laying around and the 'waterfall view' is really informative. I don't think such a complex realtime display is needed for Meshtastic but having some static UI elements that persist across views to convey airtime and channel utilization would be really useful. 

For most instances, I really do not like the emphasis on the node list (sorting through 80+ nodes is useless on non-tft nodes using the OLED or eink). A "Homepage" would find much more value IMO by displaying things like RF activity, last packet SNR/RSSI + time, battery status, bluetooth/wifi connection, GPS status, and other useful statistics or metrics that could be indicated with a glyph or graph.

**Comment by JimTheCowboy** [comment]
I started with a secondhand T-beam and progressed to T-echo's and RAK's. 
In my honest opinion there is no need for a display unless it is a standalone such as a T-deck or similar device.

The display seems to be a "nice to show gadget" with unnecessary power draw.

All necessary and interesting information can be displayed and managed via the app in a very userfriendly way.
As long as the app is required for general access and messaging I don't really see the usecase for a display of any type.
A general LED displaying on/off condition really does the job.

Connection failures or any other issues are better displayed in the app with much more information. 
So in my opinion: display - what for?

**Comment by spfmoby** [comment]
I think nodes should take 2 distinct directions:
- no screen for roof nodes, solar nodes, "always connected to smartphone" nodes... Why not one or two (configurable) LED, but it could lower the cost and power consumption of those ones.
- good quality reasonable sized touch LCD/OLED/e-paper (something around 3.5") for mobile nodes "without smartphone". even the t-deck would probably a better unit with a slightly larger/more pixels screen and no keyboard/annoying ball.
